e2e/utils.go

```diff
@@ -127,6 +127,35 @@ func updateSecretForEncryption(c kubernetes.Interface) error {
 	return nil
 }
 
+// updateVaultTokenForEncryption is an hack to update the secrets created by rook to
+// include the vault token key.
+func updateVaultTokenForEncryption(c kubernetes.Interface) error {
```

Comment:
am not sure how this feature will be integrated with the rook. Rook creates only two secrets (one for provisioner and one for node staging)  for all the users these secrets will be sent to the CSI driver even if the storageclass is not expected to be encrypted. won't there be any security problem with this?

@leseb any idea what's the plan around rook for this one?

Reply:
>am not sure how this feature will be integrated with the rook.
First of all, rook integration is independent to this repo. :) . CSI driver got its own existence with its deployment model. Rook is just one caller , there could be others too. so the secret creation from rook or its entries are not really a concern here I believe.

>Rook creates only two secrets (one for provisioner and one for node staging) for all the users these secrets will be sent to the CSI driver

Coming to the part of `secrets` will be sent to CSI driver, yes, and thats the general working mechanism for the CSI calls ( ie CO sending secret to CSI driver in many RPCs), also as long as the SC belong to the a specific driver/provisioner, thats by design and that CSI/SP driver is in charge of operating with it. Isnt it ? Also, the secret is not bound for "all" SCs and also its `not` just one. That said, different SCs can have different secrets , that well acceptable! and I would like to point that, this is `not` a new model, even we have `secret` based encryption and we pass `  # Encryption passphrase` in the same way in our implementation today!

Reply:
@humblec I know all those things, what am trying to understand is the security issue we may face and how this is planned to be integrated with rook

Reply:
@Madhu-1 again my point is that, I dont see any security concern at all and rook integration  discussion is out of the scope here atleast :+1:

Reply:
I'm not sure to fully understand the intention behind `updateVaultTokenForEncryption()`, @Madhu-1 are you referring to the secret containing the cephx keys?

Reply:
currently, for RBD rook creates  2 secrets one is `rook-csi-rbd-provisioner` and `rook-csi-rbd-node` the  secret will container cephx keys it looks like below

```yaml
"name": "rook-csi-rbd-provisioner",
"kind": "Secret",
"data": {
 "userID": 'csi-rbd-provisioner',
"userKey": PROVISIONER_KEY
}
```
with the current PR design, the vault token will be added to these secrets it will be

```yaml
"name": "rook-csi-rbd-provisioner",
"kind": "Secret",
"data": {
 "userID": 'csi-rbd-provisioner',
"userKey": PROVISIONER_KEY,
"vaultToken": XXX-XXX-XXXX
}
```
In the downstream, these are the secrets used when a user creates the storageclass. Even if the storageclass is not expected to be encrypted these secrets (created by rook)  will be sent to the CSI driver. won't there be any security problem with this?

Reply:
> @Madhu-1 again my point is that, I dont see any security concern at all and rook integration discussion is out of the scope here atleast

as am not sure about the downstream design and keeping Rook out of scope am okay